searx/engines/google_news.py

```diff
@@ -16,74 +45,155 @@
-time_range_dict = {'day': 'd',
-                   'week': 'w',
-                   'month': 'm',
-                   'year': 'y'}
-
-
-# do search-request
 def request(query, params):
-
-    search_options = {
-        'start': (params['pageno'] - 1) * number_of_results
-    }
-
-    if params['time_range'] in time_range_dict:
-        search_options['tbs'] = time_range_attr.format(range=time_range_dict[params['time_range']])
-
-    if safesearch and params['safesearch']:
-        search_options['safe'] = 'on'
-
-    params['url'] = search_url.format(query=urlencode({'q': query}),
-                                      search_options=urlencode(search_options))
-
-    if params['language'] != 'all':
-        language = match_language(params['language'], supported_languages, language_aliases).split('-')[0]
-        if language:
-            params['url'] += '&hl=' + language
+    """Google-News search request"""
+
+    language, country, lang_country = get_lang_country(
+        # pylint: disable=undefined-variable
+        params, supported_languages, language_aliases
+    )
+    subdomain = 'news.google.com'
+
+    if params['time_range']: # in time_range_dict:
+        query += ' ' + time_range_dict[params['time_range']]
+
+    query_url = 'https://'+ subdomain + '/search' + "?" + urlencode({
+        'q': query,
+        'hl': language,
+        'lr': "lang_" + language,
+        'ie': "utf8",
+        'oe': "utf8",
+        'ceid' : "%s:%s" % (country, language),
+        'gl' : country,
+    })
+
+    if params['safesearch']:
+        query_url += '&' + urlencode({'safe': filter_mapping[params['safesearch']]})
+
+    params['url'] = query_url
+    logger.debug("query_url --> %s", query_url)
+
+    # en-US,en;q=0.8,en;q=0.5
+    params['headers']['Accept-Language'] = (
+        lang_country + ',' + language + ';q=0.8,' + language + ';q=0.5'
+        )
+    logger.debug("HTTP header Accept-Language --> %s",
+                 params['headers']['Accept-Language'])
+    params['headers']['Accept'] = (
+        'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8'
+        )
+    #params['google_subdomain'] = subdomain
 
     return params
 
 
-# get response from search-request
 def response(resp):
+    """Get response from google's search request"""
     results = []
 
+    # detect google sorry
+    resp_url = urlparse(resp.url)
+    if resp_url.netloc == 'sorry.google.com' or resp_url.path == '/sorry/IndexRedirect':
+        raise RuntimeWarning('sorry.google.com')
+
+    if resp_url.path.startswith('/sorry'):
+        raise RuntimeWarning(gettext('CAPTCHA required'))
+
+    # which subdomain ?
+    # subdomain = resp.search_params.get('google_subdomain')
+
+    # convert the text to dom
     dom = html.fromstring(resp.text)
 
-    # parse results
-    for result in dom.xpath('//div[@class="g"]|//div[@class="g _cy"]'):
+    for result in eval_xpath(dom, '//div[@class="xrnccd"]'):
+
         try:
```

Comment:
@dalf what do you think, should I remove this try / except catching all exceptions?  I thinks so, otherwise SearxXPathSyntaxException and others will be catched.

Reply:
sorry I should have read https://github.com/searx/searx/pull/2483/files#r561800269 first :-)